### PR TITLE
WIP: ccxt asyncio

### PR DESCRIPTION
**Pull request by shusso**
simple proof of concept discussion opener on one possible way how we could use asyncio for ccxt

### CODE REVIEW

File: freqtrade/exchange/__init__.py [hunk 1]
```diff
@@ -286,6 +318,23 @@ def get_ticker(self, pair: str, refresh: Optional[bool] = True) -> dict:
             logger.info("returning cached ticker-data for %s", pair)
             return self._cached_ticker[pair]
 
+    async def get_ticker_history_async(self, tick_interval: str,
+                                       since_ms: Optional[int] = None):
```
Comment (misaghshakeri) [inline freqtrade/exchange/__init__.py]: where do you use since_ms ?

**Comment by creslinux** [comment]
The larger saving for tickers to get all markets/symbols in 1 requests vi CCXT unified fetch_tickers api.
For binance this is difference between 140 calls or 1, if checking all markets. 

```
print(exchange.fetch_tickers()) # all tickers indexed by their symbols
vs
print(exchange.fetch_ticker('LTC/ZEC')) # ticker for LTC/ZEC
```
The exchange info tells us if the exchange supports fetch_tickers or not
The larger exchanges tend to. 

"'
has: {                            // exchange capabilities
        'CORS': false,
        'publicAPI': true,
        'privateAPI': true,
        'cancelOrder': true,
        'createDepositAddress': false,
        'createOrder': true,
        'deposit': false,
        'fetchBalance': true,
        'fetchClosedOrders': false,
        'fetchCurrencies': false,
        'fetchDepositAddress': false,
        'fetchMarkets': true,
        'fetchMyTrades': false,
        'fetchOHLCV': false,
        'fetchOpenOrders': false,
        'fetchOrder': false,
        'fetchOrderBook': true,
        'fetchOrders': false,
        'fetchTicker': true,
        **'fetchTickers': true,**
        'fetchBidsAsks': false,
        'fetchTrades': true,
        'withdraw': false,
""

**Comment by creslinux** [comment]
For Kline requests, candle data, asyncio is a necessity as only 1 market can be got per request

**Comment by creslinux** [comment]
Ticker vs Tickers can be be best seen via a direct call to an exchange example

https://api.binance.com/api/v1/ticker/price?symbol=LTCBTC
vs this
https://api.binance.com/api/v1/ticker/price

File: freqtrade/exchange/__init__.py (hunk 1 above)
Comment (xmatthias) [reply freqtrade/exchange/__init__.py]: it's part of the original call .... so it's safe to assume that it was copied for simplicity

**Comment by xmatthias** [comment]
@shusso are you sure this is correctly using async correctly?
i would expect the calls to return "randomly" .... but it's a pure sequencial execution for me ...

**Comment by xmatthias** [comment]
i'm going to close this PR as i think this has a flaw (it's not doing async for me) - and #1101 seems to be further down the development ...

**Pull request closed without merging**